packages/sdk/src/mint/mint.test.ts

```diff
@@ -0,0 +1,178 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import { GAS, ONE_YOCTO, TOKEN_METHOD_NAMES } from '../constants';
+import { mint, MintOptions } from './mint';
+
+describe('mint method tests', () => {
+  const nftContractId = 'test.nft.contract';
+  const reference = 'test';
+  const ownerId = 'test';
+  const options = {
+    splits: { 'test': 2 },
+    amount: 2,
+    royaltyPercentage: 0.5,
+  };
+
+  const optionsWithInvalidPercentage = {
+    splits: { 'test': 2 },
+    amount: 2,
+    royaltyPercentage: 1,
+  };
+
+  const optionsWithInvalidAmount = {
+    splits: { 'test': 2 },
+    amount: 300,
+    royaltyPercentage: 0.5,
+  };
+
+  const optionsWithInvalidSplits: MintOptions =  {
+    splits:{
```

Comment:
This is quite thorough, hard to fault that but would bigger intervals work? 5ths?